### DISCUSSION TITLE
TypeError: Type is not JSON serializable: numpy.float64

### DISCUSSION THREAD

**Discussion by cordmaur** [corteva/rioxarray#474] [General]
Hi Guys,
Any reason for the methods .bounds() and .transform() return the values in `numpy.float64` instead of standard python float?
This error was being raised when creating a STAC catalog and using `rioxarray` to get footprint, bounds, etc. and the code was working fine when using plain `rasterio`. Debugging was tricky as the representation of these values is the same in the dictionary and a manual type casting was necessary for each single value.  :(

**Comment by snowman2** [comment]
I don't see anything in the code explicitly casting to `float64`. It may be something to do with being an xarray extension and the way the response gets handled under the hood.

This might be helpful: https://github.com/pyproj4/pyproj/blob/8f0e1f7ddc5ba5a1992f939f91bb9ea023316219/pyproj/utils.py#L34-L52

**Comment by cordmaur** [comment]
Thanks @snowman2. This class is helpful, but the JSON encoding is called by `pystac`. Tracking down how things work in `rioxarray`, the _internal_bounds() functions is casting the top, left, right bottom values to float (as expected). 
https://github.com/corteva/rioxarray/blob/a0322e9594dd95719bb08c988650513812457377/rioxarray/rioxarray.py#L874


The "issue" is in the _chached_transform() function:

https://github.com/corteva/rioxarray/blob/a0322e9594dd95719bb08c988650513812457377/rioxarray/rioxarray.py#L503

When calling Affine.from_gdal() and passing the np.fromstring as argument. From this point, all computations will be in numpy.float64 dtype...

![image](https://user-images.githubusercontent.com/19617404/154065984-8c059a4d-5e61-44c9-afa0-70730f91d369.png)

**Reply by snowman2** [reply]
Good catch :+1:. You are welcome to submit a PR ensuring it is a Python float.

**Reply by cordmaur** [reply]
@snowman2, PR is ready!
https://github.com/corteva/rioxarray/pull/475#issue-1140604319
